src/setup_payload/tests/BUILD.gn

```diff
@@ -34,6 +34,7 @@ chip_test_suite("tests") {
 
   public_deps = [
     "${chip_root}/src/lib/support:testing",
+    "${chip_root}/src/platform",
```

Comment:
Shouldn't this just be in `src/system/BUILD.gn`?  Seems like that's where the real dependency is...

Reply:
The issue is that `src/platform/BUILD.gn` already depends on `src/platform`. That would makes it a cyclic dependency. And the `chip::Platform::Internal::AssertChipStackLockedByCurrentThread` symbol is located in `src/platform/LockTracker.cpp`.

Reply:
#6551

> I assume you meant `src/platform/BUILD.gn` depends on `src/system`?
> 

Indeed.

> If so, just sweeping the cyclic dependency under the rug is no good either. The cyclic dependency is in fact there. I think we should either allow that explicitly in the gn file involved using `allow_circular_includes_from`, or try to figure out how to break the cycle or something.

This is not really like I'm sweeping it under the rug. It is already there since the introduction of `assertChipStackLockedByCurrentThread` in #6551 here `LockTracker.h` has been added at https://github.com/project-chip/connectedhomeip/pull/6551/files#diff-49a1739df1fe58fadd04c14fea24ba35f573ce012d2ba3e3971271c0b15ac7e2R36 and the "trick" to get that to work was to add `"${chip_root}/src/platform"`  to the reps of `src/system/tests` ( https://github.com/project-chip/connectedhomeip/pull/6551/files#diff-e2c7ed0c60c4a204d26be2f0da7d4b3385fa407029b776bd549a1f10590d1e2bR37)

I am only building on top of that, but I get your point and tried to give it a shot. Sadly I have not found any bright idea :/

Reply:
That is fair, as far as SystemLayer goes.  I think I would prefer that we just set `allow_circular_includes_from` to clearly indicate what's going on here, but I would be OK with that as a followup if you prefer...

Reply:
Let's just do a followup for that.